### PR TITLE
Change default compat level to 150

### LINKED ISSUE #34316: Change default SQL Server compat level to 150 (SQL Server 2019)
We [discussed at length](https://github.com/dotnet/efcore/issues/30163#issuecomment-1500556463) what the default should be in EF8. The last statement on this was:

> Executive architect decision: we'll default to the latest compatibility level (currently 160, SQL Server 2022). In any case, we don't expect to use any super-new feature which would block users with slightly older databases.

But then we did this #22924, which only works from level 160, which SQL Server 2022 or later. This means anyone updating to EF9 will need to either have SQL Server 2022, or set the compat level explicitly.

Instead, we are changing our previous decision and making the default 150, or SQL Server 2019. This is not a breaking change because nothing generated by EF8 requires 160.

### PR DESCRIPTION
**Pull request by ajcvickers**
Fixes #34316

### CODE REVIEW

File: test/EFCore.SqlServer.FunctionalTests/Query/ComplexNavigationsQuerySqlServer160Test.cs [hunk 1]
```diff
@@ -0,0 +1,4909 @@
...
+    public virtual async Task Distinct_skip_without_orderby(bool async)
+    {
+        await AssertQuery(
+            async,
+            ss => from l1 in ss.Set<Level1>()
+                  where l1.Id < 3
+                  select (from l3 in ss.Set<Level3>()
+                          select l3).Distinct().Skip(1).OrderBy(e => e.Id).FirstOrDefault().Name);
+
+        AssertSql(
```
Comment (roji) [inline test/EFCore.SqlServer.FunctionalTests/Query/ComplexNavigationsQuerySqlServer160Test.cs:37]: If we do keep the test suite duplication, one idea to make this lighter is to not do SQL assertions except for those tests where there's a difference

**Review by roji — APPROVED** [review]
OK, I do have an idea for how to make the tests better...

In tests where the SQL varies based on version, we could check what actual version we're running against (on TestEnvironment), and do a conditional AssertSql with the two SQLs; aside from removing the duplication, this would also show very nicely exactly where we have variations across versions...

Ideally, in CI we'd even run the tests twice - once against 2019 and once against 2022. But even if we don't (for now), we'll have 2019 in CI, and 2022 in local runs.

What do you think? Am approving as-is to not block on this, but I think it should be a pretty trivial think to do if you're up for it...

**Comment by ajcvickers** [comment]
I think we should discuss the testing some more. There are test infrastructure aspects to this. For example, we can't easily switch back and forth in a normal fixture, because the compat level is a singleton option. Also, I think there is value in asserting the SQL in both places, to catch regressions where we switch to using a 2022 feature in the 2019 cases, for example.

The main thing is that we need better support for this in the test infrastructure. And we should consider both which variations add the most value to run as well as reducing the duplication in test overrides and baselines.

**Comment by roji** [comment]
> And we should consider both which variations add the most value to run as well as reducing the duplication in test overrides and baselines

I agree, I'm just saying that we can get full coverage for both SQL variants by running builds in CI (against different SQL Server versions), rather than attempting to have both variants handled inside the same test run, via different test classes. This also would provide better cover in that we'd actually test against 2019 and 2022, as opposed to running the two SQL variants against 2022. In he PG provide, for example, we run a CI matrix against all supported versions of PG, with slight variations in SQL generation which are handled this way. 

Of course, doing this fully assumes we run both 2019 and 2022 configs in CI, which we still don't have (but I think we should go in this direction).

> For example, we can't easily switch back and forth in a normal fixture, because the compat level is a singleton option.

I don't think that's needed in what I'm proposing: the (only) test suite would check which database it's running against (by looking at TestEnvironment, which checks the actual database version), and then configure only that version in UseSqlServer(). This way each CI config (of which there would be two) would only ever execute a single SQL version against its intended actual database version.

Having said all that, I think it's perfectly fine to merge this PR as-is and clean up later!

**Pull request merged**